### PR TITLE
[3.4] Fix manifest verification if pullthrough enabled

### PR DESCRIPTION
**Pull request by legionus**
Backport of #12395.

@mfojtik @miminar @pweil-

### CODE REVIEW

**Comment by mfojtik** [comment]
hold on merging till the master branch PR is merged and QE test it.

File: pkg/dockerregistry/server/manifestschema1handler.go [hunk 1]
```diff
@@ -108,8 +108,12 @@ func (h *manifestSchema1Handler) Payload() (mediaType string, payload []byte, ca
 func (h *manifestSchema1Handler) Verify(ctx context.Context, skipDependencyVerification bool) error {
 	var errs distribution.ErrManifestVerification
 
-	// we want to verify that referenced blobs exist locally - thus using upstream repository object directly
-	repo := h.repo.Repository
+	// we want to verify that referenced blobs exist locally or accessible over
```
Comment (miminar) [inline pkg/dockerregistry/server/manifestschema1handler.go]: or are accessible via

Comment (legionus) [inline pkg/dockerregistry/server/manifestschema1handler.go]: @miminar fixed

**Comment by mfojtik** [comment]
@legionus QE finished testing, no issues found. [merge]

**Comment by mfojtik** [comment]
flake: https://github.com/openshift/origin/issues/10228

[merge]

**Pull request merged**